source/client/output_formatter_impl.cc

```diff
@@ -138,5 +144,119 @@ DottedStringOutputFormatterImpl::formatProto(const nighthawk::client::Output& ou
   return ss.str();
 }
 
+const nighthawk::client::Result&
+FortioOutputFormatterImpl::getGlobalResult(const nighthawk::client::Output& output) const {
+  for (const auto& nh_result : output.results()) {
+    if (nh_result.name() == "global") {
+      return nh_result;
+    }
+  }
+
+  throw NighthawkException("Nighthawk output was malformed, contains no 'global' results.");
+}
+
+const nighthawk::client::Counter&
+FortioOutputFormatterImpl::getCounterByName(const nighthawk::client::Result& result,
+                                            absl::string_view counter_name) const {
+  for (const auto& nh_counter : result.counters()) {
+    if (nh_counter.name() == counter_name) {
+      return nh_counter;
+    }
+  }
+
+  throw NighthawkException(absl::StrCat(
+      "Nighthawk result was malformed, contains no counter with name: ", counter_name));
+}
+
+const nighthawk::client::Statistic& FortioOutputFormatterImpl::getRequestResponseStatistic(
+    const nighthawk::client::Result& result) const {
+  for (auto const& nh_stat : result.statistics()) {
+    if (nh_stat.id() == "benchmark_http_client.request_to_response") {
+      return nh_stat;
+    }
+  }
+
+  throw NighthawkException("Nighthawk result was malformed, contains no "
+                           "'benchmark_http_client.request_to_response' statistic.");
+}
+
+std::string FortioOutputFormatterImpl::formatProto(const nighthawk::client::Output& output) const {
+  nighthawk::client::FortioResult fortio_output;
+
+  // TODO(nareddyt): Not needed but nice to have, displays in the UI
+  fortio_output.mutable_labels()->set_value("A random label");
+
+  fortio_output.mutable_starttime()->set_seconds(output.timestamp().seconds());
+  fortio_output.mutable_requestedqps()->set_value(output.options().requests_per_second().value());
+  fortio_output.mutable_requestedduration()->set_seconds(output.options().duration().seconds());
+
+  // FIXME: Set the actual to the same as the requested since NH does not expose these fields
```

Comment:
You may be able to compute the actual qps by dividing the `upstream_rq_total` counter by the specified duration.

Reply:
Thanks!